### PR TITLE
VIM-7188: Team member request + connection

### PR DESCRIPTION
**Pull request by absong1**
#### Ticket

[VIM-7188](https://vimean.atlassian.net/browse/VIM-7188)

#### Pull Request Checklist

- [x] Resolved any merge conflicts
- [x] No build errors or warnings are introduced
- [x] New files are written in Swift
- [x] New classes contain license headers
- [x] New classes have Documentation
- [x] New public methods have Documentation

#### Issue Summary

- Update networking library to support team member connection on folder as well as stand alone request to retrieve team member's for a folder.

#### Implementation Summary

- Added team members connection on `Folder` model.
- Created team member request.

#### Reviewer Tips

- n/a

#### How to Test

- n/a

### CODE REVIEW

**Comment by cameo-github** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">SwiftLint not installed, download from https://github.com/realm/SwiftLint</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 118 tests, with 0 failures (0 unexpected) in 1.779 (2.199) seconds</td>
    </tr>
  </tbody>
</table>

See build details on [CircleCI](https://circleci.com/gh/vimeo/VimeoNetworking/1430)

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Sources/Shared/Models/TeamMember.swift [hunk 1]
```diff
@@ -26,7 +26,7 @@
 
 import Foundation
 
-public class TeamMember: VIMModelObject, ConnectionsProviding, ConnectionsParsing {
+@objcMembers public class TeamMember: VIMModelObject, ConnectionsProviding, ConnectionsParsing {
```
Comment (bryantjustin) [inline Sources/Shared/Models/TeamMember.swift]: I think it's usually placed in a new line? Not 100% sure. But that's how I've seen it in the past with other classes.

File: Sources/Shared/Request+User.swift [hunk 2]
```diff
@@ -131,4 +131,17 @@ public extension Request {
     static func patchUser(withUserURI userURI: String, parameters: VimeoClient.RequestParametersDictionary) -> Request {
         return Request(method: .PATCH, path: userURI, parameters: parameters)
     }
+
```
Comment (bryantjustin) [inline Sources/Shared/Request+User.swift]: It's also fine if this was in the other file.

**Review by bryantjustin — APPROVED** [review]

**Comment by asongvimeo** [comment]
build this please

**Comment by asongvimeo** [comment]
build this please

**Pull request merged**